packages/components/src/components/domainproperties/list/models.ts

```diff
@@ -103,6 +103,14 @@ export class ListModel extends Record({
             return new ListModel({...defaultSettings, domain});
         } else {
             let domain = DomainDesign.create(raw.domainDesign);
+
+            // Issue39818: Set the key field of an existing list to be PKLocked.
+            const fields = domain.fields;
+            const pkField = fields.findIndex(i => (i.isPrimaryKey));
+            const pkFieldLocked = fields.get(pkField).merge({ lockType:"PKLocked" }) as DomainField;
```

Comment:
I assume it isn't possible for the pkField to not exist for a ListModel at this point of the code, right? Would it be worth adding in a check for `if (pkFieldIndex > -1)` or something along those lines here before we start updating the fields?